app/src/main/java/fr/neamar/kiss/adapter/RecordAdapter.java

```diff
@@ -61,13 +68,61 @@ public View getView(int position, View convertView, ViewGroup parent) {
         return results.get(position).display(getContext(), results.size() - position, convertView);
     }
 
-    public void onLongClick(int pos) {
+    public void onLongClick(final int pos, View v) {
+        // Check that the pojo is an app
+        if (getItemViewType(pos) == 0)
+        {
+            final AppPojo appPojo = ((AppResult)results.get(pos)).appPojo;
+            PopupMenu popup = new PopupMenu(getContext(), v);
+            popup.getMenuInflater().inflate(R.menu.menu_app, popup.getMenu());
+
+            popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+                public boolean onMenuItemClick(MenuItem item) {
+                    popupMenuClickHandler(appPojo, pos, item);
+                    return true;
+                }
+            });
+
+            popup.show();
+        }
+        else {
+            removeItem(pos);
+        }
+    }
+
+    private void popupMenuClickHandler(AppPojo appPojo, int pos, MenuItem item) {
```

Comment:
(maybe you can have a default behavior on Result class to handle the item_remove action)